pypeit/core/flexure.py

```diff
@@ -217,56 +211,59 @@ def spec_flex_shift(obj_skyspec, arx_skyspec, arx_fwhm_pix, spec_fwhm_pix=None,
     obj_skyspec.data['flux'][0,:] = np.maximum(obj_skyspec.data['flux'][0,:], 0.)
     arx_skyspec.data['flux'][0,:] = np.maximum(arx_skyspec.data['flux'][0,:], 0.)
 
+    # clip too large values (>90%) only in obj_skyspec (assuming arx_skyspec is being vetted before)
+    # this is used ony for the cross-correlation
+    obj_skyspec_flux = obj_skyspec.flux.value
+    _lower = np.percentile(obj_skyspec_flux[obj_skyspec_flux < 0.0], 90) if np.any(obj_skyspec_flux < 0.0) else 0.0
+    _upper = np.percentile(obj_skyspec_flux[obj_skyspec_flux >= 0.0], 90) if np.any(obj_skyspec_flux >= 0.0) else 0.0
+    obj_skyspec_flux = np.clip(obj_skyspec_flux, _lower, _upper)
```

Comment:
You perform a similar operation a few times, it's likely worth making a short function.